src/controllers/ClinicController.java

```diff
@@ -22,11 +26,13 @@ public class ClinicController extends TerminalController {
      * @param previousController UserController<Admin> - the object of the admin controller that switched into this
      *                           clinic controller object.
      */
-    public ClinicController(Context context, UserController<Admin> previousController) {
+    public ClinicController(Context context, UserController<Admin> previousController, UserData<Admin> adminData) {
```

Comment:
why is this UserData<Admin> adminData. It should be AdminData adminData. AdminData is more specific then UserData<Admin>